File: boat-spring-boot-sample/src/test/java/sample/java/xyz/srclab/spring/boot/core/BeanSample.java
package sample.java.xyz.srclab.spring.boot.core;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.context.annotation.DependsOn;
import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
import org.testng.Assert;
import org.testng.annotations.Test;

import javax.annotation.Resource;
import java.util.Arrays;

@SpringBootTest(classes = Starter.class)
//@ContextConfiguration(classes = {TestStarter.class})
@DependsOn("myBeanRegistry")
public class BeanSample extends AbstractTestNGSpringContextTests {

    private static final Logger logger = LoggerFactory.getLogger(BeanSample.class);

    @Resource
    private MyBeanLifecyclePostProcessor myBeanLifecyclePostProcessor;

    @Resource
    private String bean1;

    @Resource
    private String bean2;

    @Resource
    private MyBean myBean;

    @Test
    public void testBeanPostProcessor() {
        logger.info("Bean processing sequence: {}", myBeanLifecyclePostProcessor.getSequence());
        Assert.assertEquals(
                myBeanLifecyclePostProcessor.getSequence(),
                Arrays.asList(
                        "postProcessBeanDefinitionRegistry",
                        "postProcessBeanFactory",
                        "postProcessBeforeInstantiation",
                        "postProcessAfterInstantiation",
                        "postProcessProperties",
                        "postProcessBeforeInitialization",
                        "postProcessAfterInitialization"
                )
        );
    }

    @Test
    public void testBeanManager() {
        logger.info("bean1: {}", bean1);
        Assert.assertEquals(bean1, "bean1");
        logger.info("bean2: {}", bean2);
        Assert.assertEquals(bean2, "bean2");
        logger.info("myBean: {}", myBean.getBeanString());
        Assert.assertEquals(myBean.getBeanString(), bean1 + bean2);
    }
}

File: boat-spring-boot-sample/src/test/kotlin/sample/kotlin/xyz/srclab/spring/boot/core/BeanSample.kt
package sample.kotlin.xyz.srclab.spring.boot.core

import org.slf4j.LoggerFactory
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.context.annotation.DependsOn
import org.springframework.test.context.testng.AbstractTestNGSpringContextTests
import org.testng.Assert
import org.testng.annotations.Test
import javax.annotation.Resource

@SpringBootTest(classes = [Starter::class])
//@ContextConfiguration(classes = [TestStarter::class])
@DependsOn("myBeanRegistryKt")
open class BeanSample : AbstractTestNGSpringContextTests() {

    @Resource
    private lateinit var myBeanLifecyclePostProcessor: MyBeanLifecyclePostProcessor

    @Resource
    private lateinit var bean1: String

    @Resource
    private lateinit var bean2: String

    @Resource
    private lateinit var myBean: MyBean

    @Test
    fun testBeanPostProcessor() {
        log.info("Bean processing sequence: {}", myBeanLifecyclePostProcessor.getSequence())
        Assert.assertEquals(
            myBeanLifecyclePostProcessor.getSequence(),
            listOf(
                "postProcessBeanDefinitionRegistry",
                "postProcessBeanFactory",
                "postProcessBeforeInstantiation",
                "postProcessAfterInstantiation",
                "postProcessProperties",
                "postProcessBeforeInitialization",
                "postProcessAfterInitialization"
            )
        )
    }

    @Test
    fun testBeanManager() {
        log.info("bean1: {}", bean1)
        Assert.assertEquals(bean1, "bean1")
        log.info("bean2: {}", bean2)
        Assert.assertEquals(bean2, "bean2")
        log.info("myBean: {}", myBean.beanString)
        Assert.assertEquals(myBean.beanString, bean1 + bean2)
    }

    companion object {
        private val log = LoggerFactory.getLogger(BeanSample::class.java)
    }
}

@SpringBootApplication
open class Starter